### PR TITLE
[NON-MODULAR] fixes simple mob emote spam in dchat

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

credits to azarak for the idea :3

approved by xyel, since it's a QoL

non-modular, because of the feature freeze upstream, and QoLs would get closed there

## Why It's Good For The Game

my chat won't be 40-50% simple mobs after this

## Changelog
:cl:
qol: no more simple mob emote spam in dchat
/:cl:

### CODE REVIEW

File: code/datums/emotes.dm [hunk 1]
```diff
@@ -121,11 +121,12 @@
 		//SKYRAT EDIT CHANGE END
 
 	var/user_turf = get_turf(user)
-	for(var/mob/ghost in GLOB.dead_mob_list)
-		if(!ghost.client || isnewplayer(ghost))
-			continue
-		if(ghost.stat == DEAD && ghost.client && user.client && (ghost.client.prefs.chat_toggles & CHAT_GHOSTSIGHT) && !(ghost in viewers(user_turf, null)))
-			ghost.show_message("<span class='emote'>[FOLLOW_LINK(ghost, user)] [dchatmsg]</span>")
+	if(user.client)
+		for(var/mob/ghost in GLOB.dead_mob_list)
```
Comment (ZephyrTFA) [inline code/datums/emotes.dm:125]:
```suggestion
		for(var/mob/ghost as anything in GLOB.dead_mob_list)
```

File: code/datums/emotes.dm [hunk 2]
```diff
@@ -301,10 +302,11 @@
 	var/ghost_text = "<b>[src]</b> [text]"
 
 	var/origin_turf = get_turf(src)
-	for(var/mob/ghost in GLOB.dead_mob_list)
-		if(!ghost.client || isnewplayer(ghost))
-			continue
-		if(ghost.stat == DEAD && ghost.client && (ghost.client.prefs.chat_toggles & CHAT_GHOSTSIGHT) && !(ghost in viewers(origin_turf, null)))
-			ghost.show_message("[FOLLOW_LINK(ghost, src)] [ghost_text]")
+	if (client)
+		for(var/mob/ghost in GLOB.dead_mob_list)
```
Comment (ZephyrTFA) [inline code/datums/emotes.dm:306]:
```suggestion
		for(var/mob/ghost as anything in GLOB.dead_mob_list)
```

File: code/datums/emotes.dm (hunk 1 above)
Comment (Seris02) [reply code/datums/emotes.dm:125]: but why
that's just editing more code from tg to be non-modular
what would that even do

**Comment by Gandalf2k15** [comment]
Modular comments

**Comment by Gandalf2k15** [comment]
This should really be done upstream actually.

**Comment by Gandalf2k15** [comment]
upstreams

**Pull request closed without merging**